### ISSUE TITLE
Dynamic grid views

### ISSUE THREAD

**Issue by constantinpape** [mobie/mobie-viewer-fiji#362]
Currently, grid views need to be specified in the project metadata ahead of time. For many applications it would be useful to enable dynamic grid views for a user selection of sources.
I could think about different options for this:
- Add these via a menu where sources can be selected from a list of all sources.
- Add these via a menu where sources are selected via name wildcards, e.g. `nuclei_*`
- Add option to add next view as grid instead of at same location upon clicking the `view` button.

**Comment by K-Meech** [comment]
This would be great! This would also make it easier to generate grid view bookmarks via 'save current settings as view...' in the context menu

**Comment by constantinpape** [comment]
@K-Meech do you have time to work on this in the next few weeks? Otherwise it may also be a good task for @KateMoreva after the refactoring of the n5/ome-zarr code is done.

**Comment by K-Meech** [comment]
I won't have too much time - so @KateMoreva, go ahead! Once the grid view ui stuff is in place, I can make it work with the rest of the bookmark code :)

**Comment by constantinpape** [comment]
> I won't have too much time - so @KateMoreva, go ahead! Once the grid view ui stuff is in place, I can make it work with the rest of the bookmark code :)

Ok! @KateMoreva we can discuss details on this once you are done with the refactoring of n5/zarr readers :).

**Comment by constantinpape** [comment]
To make this more concrete: the idea here is to add some UI functionality to create grid views of multiple sources on-the-fly.
I think a good first version of this functionality would be:
- Add a simple menu where a wildcard pattern can be entered, e.g. `nuclei_*`, to select the sources for the grid view. (This UI can be improved later).
- Create the corresponding view by specifying the correct `gridSourceTransform` and `sourceDisplays`.

cc @KateMoreva, @K-Meech

**Comment by KateMoreva** [comment]
I think there are two main variants for the menu:

1 (working on this variant)
![image](https://user-images.githubusercontent.com/48569608/130196709-6f29fbfa-9421-40ad-8690-e1afa6774ec9.png)
Sources settings can be set through the different menus by click on 3 vertical dots. The grid view flag is visible in the main menu, sources can be added from both the main and this menu.
+ Easier to implement and add new functionality if needed 
- Info is a bit "hidden" from the user 

2
![image](https://user-images.githubusercontent.com/48569608/130228904-65d26804-40c0-4656-bad9-e65675663ece.png)
The grid view creation button is on the main page
+ All info is in one place 
- Harder to implement as old logic needs to be changed 


@constantinpape, @K-Meech,  @tischi what do you think?

**Comment by K-Meech** [comment]
Hi @KateMoreva - I think tischi and Constantin are on holiday this week, so you'll have to wait until Monday to get their opinions.

Looking good so far! I think variant 1 will be the easiest to make + maintain. For now it would be easiest to access this through the right click context menu in the bigdataviewer window. (This is where all the other extra menus e.g. bookmark creation etc are stored for now)
Variant 2 would be nice - but I think getting it up and running with variant 1 first makes a lot of sense :) We can always add this in future once everything's working!

**Comment by KateMoreva** [comment]
> Looking good so far! I think variant 1 will be the easiest to make + maintain. For now it would be easiest to access this through the right click context menu in the bigdataviewer window. (This is where all the other extra menus e.g. bookmark creation etc are stored for now)

Ok, thank you, so far I added an extra button for creating grid view, but if users are used to working with the right-click window then I will move it there.

**Comment by K-Meech** [comment]
Ok - sounds good! Actions are added to the right-click/context menu here: https://github.com/mobie/mobie-viewer-fiji/blob/1eb49182daf3adc718739dd5c5b958ad67be5352/src/main/java/de/embl/cba/mobie/bdv/view/SliceViewer.java#L73

**Comment by constantinpape** [comment]
I agree with @K-Meech that adding this to the context menu (= right click) is the best option for now. We can think about including it in some other parts of the GUI later once we have a better idea of how to handle this from the UI side.
